source/Octopus.Tentacle.Tests/Contracts/ScriptTicketFixture.cs

```diff
@@ -0,0 +1,32 @@
+﻿using FluentAssertions;
+using NUnit.Framework;
+using Octopus.Tentacle.Contracts;
+
+namespace Octopus.Tentacle.Tests.Contracts
+{
+    [TestFixture]
+    public class ScriptTicketFixture
+    {
+        [Test]
+        public void GetHashCodeShouldIgnoreCasing()
+        {
+            var originalScriptTicket = new ScriptTicket("MixedCasing");
+            var upperCaseScriptTicket = new ScriptTicket(originalScriptTicket.TaskId.ToUpper());
+            var lowerCaseScriptTicket = new ScriptTicket(originalScriptTicket.TaskId.ToLower());
+
+            upperCaseScriptTicket.GetHashCode().Should().Be(originalScriptTicket.GetHashCode());
+            lowerCaseScriptTicket.GetHashCode().Should().Be(originalScriptTicket.GetHashCode());
+        }
+
+        [Test]
+        public void EqualityShouldIgnoreCasing()
+        {
+            var originalScriptTicket = new ScriptTicket("MixedCasing");
+            var upperCaseScriptTicket = new ScriptTicket(originalScriptTicket.TaskId.ToUpper());
+            var lowerCaseScriptTicket = new ScriptTicket(originalScriptTicket.TaskId.ToLower());
+
+            upperCaseScriptTicket.Should().Be(originalScriptTicket);
```

Comment:
Since ScriptTicket has a custom equality implementation, I think we should be explicitly testing all the ways equality can be tested.

This test relies on whichever equality test `FluentAssertions` is choosing to use. It would be easier to see if we are covering all required cases if we had an 'act' part that used the explicit method (`Equals(...)`, `==`, `!=`, etc.), and then assert on the result.

Reply:
Good call. Fixed!